server/src-lib/Hasura/Server/Init.hs

```diff
@@ -959,15 +959,36 @@ parseLogLevel = optional $
 
 -- Init logging related
 connInfoToLog :: Q.ConnInfo -> StartupLog
-connInfoToLog (Q.ConnInfo host port user _ db _ retries) =
+connInfoToLog ci =
   StartupLog L.LevelInfo "postgres_connection" infoVal
   where
-    infoVal = J.object [ "host" J..= host
-                       , "port" J..= port
-                       , "user" J..= user
-                       , "database" J..= db
-                       , "retries" J..= retries
-                       ]
+    infoVal = case ci of
+      Q.CIDatabaseURI retries uri -> mkDBUriLog retries $
+                                     T.unpack $ bsToTxt uri
+      Q.CIOptions co              ->
+        J.object [ "host" J..= Q.connHost co
+                 , "port" J..= Q.connPort co
+                 , "user" J..= Q.connUser co
+                 , "database" J..= Q.connDatabase co
+                 , "retries" J..= Q.connRetries co
+                 ]
+
+    mkDBUriLog retries uri =
+      let mUri = parseURI uri
+          mParsedUri = flip fmap mUri $ \u -> uriToString hidePass u ""
```

Comment:
The `Show` instance implementation of `URI` already hides password (see docs of `uriToString`) [1]. We should directly use it instead of our own implementation.


[1]: https://hackage.haskell.org/package/network-uri-2.6.1.0/docs/Network-URI.html#g:6
[2]: http://hackage.haskell.org/package/network-uri-2.6.1.0/docs/src/Network-URI.html#line-211